mypy/types.py

```diff
@@ -2587,7 +2587,7 @@ class EllipsisType(ProperType):
 
     def accept(self, visitor: TypeVisitor[T]) -> T:
         assert isinstance(visitor, SyntheticTypeVisitor)
-        return visitor.visit_ellipsis_type(self)
+        return cast(T, visitor.visit_ellipsis_type(self))
```

Comment:
How do you decide between `cast` and `assert`? 🤔

Reply:
I would generally use `assert` wherever possible, as it provides runtime verification of the type narrowing. `cast` feels closer to telling the type checker to shut up and go away 😄

But here, we can't use `assert` — how can we assert that something is of type `T`, when we don't know what `T` is (since `T` is a `TypeVar`)? And this also feels like somewhere where, ideally, mypy _should_ be able to infer that the returned type is of type `T` without any manual type narrowing. So in this specific situation, I'm more at ease with telling the type checker to shut up and go away 😉